test/e2e/setup.go

```diff
@@ -0,0 +1,367 @@
+				},
+				APIVersion: apiVersions["network"],
+			},
+		},
+	}
+	c.log.Infof("creating vnet")
+	err = c.deployments.CreateOrUpdateAndWait(ctx, c.rg, "azuredeploy", azresources.Deployment{
+		Properties: &azresources.DeploymentProperties{
+			Template: t,
+			Mode:     azresources.Incremental,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	c.log.Debugf("assigning role to AZURE_FP_CLIENT_ID for vnet")
+	err = c.assignVnetRoleTo(ctx, os.Getenv("AZURE_FP_CLIENT_ID"))
+	if err != nil {
+		return err
+	}
+	c.log.Debugf("assigning role to AZURE_CLUSTER_CLIENT_ID for vnet")
+	err = c.assignVnetRoleTo(ctx, os.Getenv("AZURE_CLUSTER_CLIENT_ID"))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *clientSet) getKubeAdminPassword(ctx context.Context, clusterName string) error {
+	c.log.Info("getting kube admin password")
+	cred, err := c.openshiftclusters.GetCredentials(ctx, clusterName, clusterName)
+	if err != nil {
+		return err
+	}
+	if cred.KubeadminPassword != nil {
+		kubeadminPassword = *cred.KubeadminPassword
+	} else {
+		c.log.Warnf("KubeadminPassword is empty")
+	}
+	return nil
+}
+
+func (c *clientSet) createCluster(ctx context.Context, vnetRG, clusterName string) error {
+	b, err := ioutil.ReadFile("cluster.json")
+	if err != nil {
+		return err
+	}
+
+	oc := redhatopenshift.OpenShiftCluster{}
+	err = oc.UnmarshalJSON(b)
+	if err != nil {
+		return err
+	}
+	c.log.Infof("creating cluster %s", clusterName)
+	future, err := c.openshiftclusters.Create(ctx, clusterName, clusterName, oc)
+	if err != nil {
+		return err
+	}
+
+	c.log.Infof("waiting for cluster %s", clusterName)
+	return future.WaitForCompletionRef(ctx, c.openshiftclusters.Client)
+}
+
+var _ = AfterSuite(func() {
+	logger := utillog.GetTestLogger()
+	cs, err := newClientSet(
+		logger,
+		os.Getenv("LOCATION"),
+		os.Getenv("AZURE_SUBSCRIPTION_ID"),
+		os.Getenv("AZURE_TENANT_ID"),
+		os.Getenv("VNET_RESOURCEGROUP"))
+	if err != nil {
+		logger.Fatal(err)
+		return
+	}
+	if os.Getenv("E2E_NO_DELETE") == "true" {
+		logger.Warn("E2E_NO_DELETE set, not deleting the test cluster")
+		return
+	}
+
+	ctx := context.Background()
+	logger.Infof("AfterSuite deleting cluster %s", os.Getenv("CLUSTER"))
+	future, err := cs.openshiftclusters.Delete(ctx, os.Getenv("CLUSTER"), os.Getenv("CLUSTER"))
+	if err != nil {
+		logger.Error(err)
+	}
+	err = future.WaitForCompletionRef(ctx, cs.openshiftclusters.Client)
+	if err != nil {
+		logger.Error(err)
+	}
+	logger.Infof("AfterSuite deleting vnet %s", cs.rg)
+	err = cs.deployments.DeleteAndWait(ctx, cs.rg, "azuredeploy")
+	if err != nil {
+		logger.Error(err)
+	}
+	logger.Infof("AfterSuite deleting vnet resource group %s", cs.rg)
+	err = cs.groups.DeleteAndWait(ctx, cs.rg)
+	if err != nil {
+		logger.Error(err)
+	}
+})
+
+var _ = BeforeSuite(func() {
+	logger := utillog.GetTestLogger()
+	for _, key := range []string{
+		"LOCATION", "AZURE_SUBSCRIPTION_ID", "AZURE_TENANT_ID",
+		"CLUSTER", "VNET_RESOURCEGROUP",
```

Comment:
Maybe e2e should pick its own CLUSTER and RESOURCEGROUP and use RESOURCEGROUP instead of VNET_RESOURCEGROUP